src/hyperswarm-dht-logger.js

```diff
@@ -0,0 +1,79 @@
+'use strict'
+
+const debug = require('debug')('hcore-logger')
+const DHT = require('@hyperswarm/dht')
+const Tail = require('tail').Tail
+const { crypto_sign_SEEDBYTES } = require('sodium-universal') // eslint-disable-line
+
+class HyperSwarmDHTLogger {
+  /**
+   * @param {string} file
+   * @param {string|null} seed
+   */
+  constructor (file, seed = null) {
+    this.file = file
+    this.seed = seed && Buffer.allocUnsafe(crypto_sign_SEEDBYTES).fill(seed)
+
+    this.sockets = []
+    this.node = null
+    this.sever = null
+    this.fileWatcher = null
+  }
+
+  async start () {
+    this.node = new DHT()
+    this.server = this.node.createServer()
+
+    this.server.on('connection', async socket => {
+      debug('socket connected')
+      this.addSocket(socket)
```

Comment:
I would suggest to use either Set or Map<string, Socket> to handle client sockets
this could be e.g.
```js
      const id = uuid.v4()
      debug('socket %s connected', id)
      this.addSocket(id, socket)

      socket.on('error', error => {
        debug('socket connection %s', error)
        this.removeSocket(id)
        debug('socket %s ended', id)
      })

      socket.on('end', () => {
        this.removeSocket(id)
        debug('socket %s ended', id)
      })
```
```js
  addSocket (id, socket) {
    this.sockets.set(id, socket)
  }

  removeSocket (socket) {
    const socket = this.sockets.get(id)
    if (socket) {
       this.sockets.del(id)
       socket.end()
       socket.destroy()
    }
  }
```
Reason why I'm suggesting maps is because of instant access instead of iterating over elements